bionic/core/task_execution.py

```diff
@@ -751,13 +752,35 @@ def _local_artifact_from_value(self, value):
                 )
             ) from e
 
-        value_hash = protocol.tokenize_file(value_path)
+        value_hash = self._generate_value_hash(value, value_path, context)
         return Artifact(
             url=url_from_path(value_path),
             content_hash=value_hash,
         )
         return value_path
 
+    def _generate_value_hash(self, value, value_path, context):
+        protocol = self.desc_metadata.protocol
+        code_versioning_policy = self.func_attrs.code_versioning_policy
+        code_version = code_versioning_policy.version
+        versioning_policy = context.core.versioning_policy
+        try:
+            bytecode_hash = CodeHasher.hash(
+                type(value),
+                code_versioning_policy.suppress_bytecode_warnings
+                or versioning_policy.ignore_bytecode_exceptions
+                or not code_version.includes_bytecode,
```

Comment:
Hmm... if `not code_version.includes_bytecode`, should we be doing this hash at all?

Reply:
I'm not 100% sure what the right thing to do here is either. Let's discuss this when we come back to work.

Reply:
I'm skipping it for now, but open to discussion on this.